components/renku_data_services/solr/solr_migrate.py

```diff
@@ -132,13 +136,36 @@ class MigrationState(BaseModel):
     skipped_migrations: int
 
 
+class SolrMigrateException(Exception):
+    """Base exception for migration errors."""
+
+    def __init(self, message: str) -> None:
+        super().__init__(message)
+
+
 class SchemaMigrator:
     """Allows to inspect the current schema version and run schema migrations against a solr core."""
 
     def __init__(self, cfg: SolrClientConfig) -> None:
         self.__config = cfg
         self.__docId: str = "VERSION_ID_EB779C6B-1D96-47CB-B304-BECF15E4A607"
 
+    async def ensure_core(self) -> None:
+        """Ensures an existing core.
+
+        If no core is found, one is created using the admin api.
+        """
+        async with DefaultSolrAdminClient(self.__config) as client:
+            status = await client.status(None)
```

Comment:
nitpick: It may "sound" a bit better if the method is called `core_status` instead of just status. But dont worry this is fine. Feel free to ignore this.